fix potsanity looking at deku tree MQness when dealing with pots between quests

serprex opened the pull request:
also don't need a loop for a range check

<!--- section:artifacts:start -->
### Build Artifacts
  - [soh-mac.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/2372917294.zip)
  - [soh-linux.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/2372922952.zip)
  - [soh-windows.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/2372924920.zip)
<!--- section:artifacts:end -->

serprex committed:
fix potsanity looking at deku tree MQness when dealing with pots between dungeons

also don't need a loop for a range check

serprex committed:
share logic between placing freestanding & pots

serprex commented:
decided to scope creep on this PR to not just the potsanity typo, but also the free standing code looking weird, to quote myself from Discord:

> allLocations not include dungeons? unsure why these two loops run unconditionally, feels like in shuffle all dungeons would double dip. Other code in this file tends to loop over allLocations & just do loc->IsDungeon() in single loop

Malkierian commented:
I love the idea of consolidation, but I don't think that check at the beginning of `PlaceItemsForType` for `overworldActive && dungeonActive` should be there at all. you're checking for overworld and dungeon separately later, and that also gatekeeps the entire block behind both being true, which isn't always the case.

Looks like you mistyped a lot of enums as well, all builds are failing. Did you build locally before pushing to test?

serprex committed:
fix settings typo

serprex commented:
Kicked off building on my end after pushing. Seeing all freestanding items being shuffled when doing dungeons only. Looking into

Good catch, I'm haunted by constantly flipping things in my boolean expressions unfortunately

serprex committed:
fix all freestanding items still being shuffled

this reinstates original freestanding method of filling in unused locations,
but also extend logic so that when only dungeons have freestanding shuffled
we don't render overworld freestanding as 3d versions until collected

Malkierian commented:
We desperately need to unify the method of adding/excluding based on vanilla status...

Malkierian commented:
The only other thing I want to bring up for discussion is whether the check for `overworldActive || dungeonActive` should be done in the function or before the call to the function. Honestly not sure which would be better practice, but I feel like checking before calling would make the code flow clearer.

serprex committed:
lift optimization check
